plugins/modules/s3_bucket.py

```diff
@@ -725,6 +765,35 @@ def put_bucket_encryption(s3_client, bucket_name, encryption):
     server_side_encryption_configuration = {'Rules': [{'ApplyServerSideEncryptionByDefault': encryption}]}
     s3_client.put_bucket_encryption(Bucket=bucket_name, ServerSideEncryptionConfiguration=server_side_encryption_configuration)
 
+def put_bucket_key_with_retry(module, s3_client, name, expected_encryption):
+    max_retries = 3
+    for retries in range(1, max_retries + 1):
+        try:
```

Comment:
It shouldn't be necessary to call put_bucket_encryption multiple times, I suspect the waiter just needs to be more patient, sometimes settings do take a couple of minutes to propagate.